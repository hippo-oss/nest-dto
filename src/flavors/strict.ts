import { composeDecoratorFactories } from '@hippo-oss/dto-decorators';
import { CLASS_DECORATORS } from '@hippo-oss/class-decorators';

export const DECORATORS = composeDecoratorFactories([
    CLASS_DECORATORS,
]);

export const {
    IsBoolean,
    IsDate,
    IsDateString,
    IsEnum,
    IsInteger,
    IsNested,
    IsNumber,
    IsString,
    IsUUID,
} = DECORATORS;
