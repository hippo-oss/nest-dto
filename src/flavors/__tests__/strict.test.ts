import { plainToClass } from 'class-transformer';
import { validate } from 'class-validator';

import * as strict from '../strict';
import { INPUT, createFixtures } from './fixtures';

describe('flavors.strict', () => {
    const Example = createFixtures(strict);

    it('validates required fields', async () => {
        const obj = plainToClass(Example, {});

        // expect every field to exist, but be undefined
        expect(Object.keys(obj)).toHaveLength(36);
        for (const value of Object.values(obj)) {
            expect(value).toBeUndefined();
        }
        expect(obj).toMatchSnapshot();

        const errors = await validate(obj);
        // we expect an error for every required field (but not the optional ones)
        expect(errors).toHaveLength(14);
        expect(errors).toMatchSnapshot();
    });
    it('transforms input data', async () => {
        const obj = plainToClass(Example, INPUT);

        // expect all data to be transformed
        expect(Object.keys(obj)).toHaveLength(36);
        expect(obj).toMatchObject({
            falseBooleanValue: false,
            zeroBooleanValue: false,
            requiredObjectArrayValue: [{
                requiredStringValue: 'nested',
            }],
            requiredPrimitiveArrayValue: [
                42,
            ],
            requiredNestedValue: {
                requiredStringValue: 'nested',
            },
            optionalObjectArrayValue: [{
                optionalStringValue: 'nested',
            }],
            optionalPrimitiveArrayValue: [
                42,
            ],
            optionalNestedValue: {
                optionalStringValue: 'nested',
            },
            nullableObjectArrayValue: [{
                nullableStringValue: 'nested',
            }],
            nullablePrimitiveArrayValue: [
                42,
            ],
            nullableNestedValue: {
                nullableStringValue: 'nested',
            },
            stringValueWithDifferentWireName: 'value',
        });

        const errors = await validate(obj);
        // we expect no errors
        expect(errors).toHaveLength(0);
        expect(errors).toMatchSnapshot();
    });
    it('transforms input data if extraneous values are excluded', async () => {
        const obj = plainToClass(Example, INPUT, {
            excludeExtraneousValues: true,
        });

        // expect all data to be transformed (b/c @Expose() is included)
        expect(Object.keys(obj)).toHaveLength(36);

        const errors = await validate(obj);
        // we expect no errors
        expect(errors).toHaveLength(0);
        expect(errors).toMatchSnapshot();
    });
});
