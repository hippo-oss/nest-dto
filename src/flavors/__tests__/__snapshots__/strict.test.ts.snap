// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flavors.strict transforms input data 1`] = `Array []`;

exports[`flavors.strict transforms input data if extraneous values are excluded 1`] = `Array []`;

exports[`flavors.strict validates required fields 1`] = `
Example {
  "falseBooleanValue": undefined,
  "nullableBooleanValue": undefined,
  "nullableDateStringValue": undefined,
  "nullableDateValue": undefined,
  "nullableEnumValue": undefined,
  "nullableIntegerValue": undefined,
  "nullableNestedValue": undefined,
  "nullableNumberValue": undefined,
  "nullableObjectArrayValue": undefined,
  "nullablePrimitiveArrayValue": undefined,
  "nullableStringValue": undefined,
  "nullableUUIDValue": undefined,
  "optionalBooleanValue": undefined,
  "optionalDateStringValue": undefined,
  "optionalDateValue": undefined,
  "optionalEnumValue": undefined,
  "optionalIntegerValue": undefined,
  "optionalNestedValue": undefined,
  "optionalNumberValue": undefined,
  "optionalObjectArrayValue": undefined,
  "optionalPrimitiveArrayValue": undefined,
  "optionalStringValue": undefined,
  "optionalUUIDValue": undefined,
  "requiredBooleanValue": undefined,
  "requiredDateStringValue": undefined,
  "requiredDateValue": undefined,
  "requiredEnumValue": undefined,
  "requiredIntegerValue": undefined,
  "requiredNestedValue": undefined,
  "requiredNumberValue": undefined,
  "requiredObjectArrayValue": undefined,
  "requiredPrimitiveArrayValue": undefined,
  "requiredStringValue": undefined,
  "requiredUUIDValue": undefined,
  "stringValueWithDifferentWireName": undefined,
  "zeroBooleanValue": undefined,
}
`;

exports[`flavors.strict validates required fields 2`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isArray": "requiredObjectArrayValue must be an array",
      "isDefined": "requiredObjectArrayValue should not be null or undefined",
    },
    "property": "requiredObjectArrayValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isArray": "requiredPrimitiveArrayValue must be an array",
      "isDefined": "requiredPrimitiveArrayValue should not be null or undefined",
      "isNumber": "each value in requiredPrimitiveArrayValue must be a number conforming to the specified constraints",
    },
    "property": "requiredPrimitiveArrayValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "falseBooleanValue must be a boolean value",
      "isDefined": "falseBooleanValue should not be null or undefined",
    },
    "property": "falseBooleanValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "zeroBooleanValue must be a boolean value",
      "isDefined": "zeroBooleanValue should not be null or undefined",
    },
    "property": "zeroBooleanValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "requiredBooleanValue must be a boolean value",
      "isDefined": "requiredBooleanValue should not be null or undefined",
    },
    "property": "requiredBooleanValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "requiredDateStringValue should not be null or undefined",
      "isIso8601": "requiredDateStringValue must be a valid ISO 8601 date string",
    },
    "property": "requiredDateStringValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDate": "requiredDateValue must be a Date instance",
      "isDefined": "requiredDateValue should not be null or undefined",
    },
    "property": "requiredDateValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "requiredEnumValue should not be null or undefined",
      "isEnum": "requiredEnumValue must be a valid enum value",
    },
    "property": "requiredEnumValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "requiredIntegerValue should not be null or undefined",
      "isInt": "requiredIntegerValue must be an integer number",
    },
    "property": "requiredIntegerValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "requiredNestedValue should not be null or undefined",
    },
    "property": "requiredNestedValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "requiredNumberValue should not be null or undefined",
      "isNumber": "requiredNumberValue must be a number conforming to the specified constraints",
    },
    "property": "requiredNumberValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "requiredStringValue should not be null or undefined",
      "isString": "requiredStringValue must be a string",
    },
    "property": "requiredStringValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "requiredUUIDValue should not be null or undefined",
      "isUuid": "requiredUUIDValue must be a UUID",
    },
    "property": "requiredUUIDValue",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "stringValueWithDifferentWireName should not be null or undefined",
      "isString": "stringValueWithDifferentWireName must be a string",
    },
    "property": "stringValueWithDifferentWireName",
    "target": Example {
      "falseBooleanValue": undefined,
      "nullableBooleanValue": undefined,
      "nullableDateStringValue": undefined,
      "nullableDateValue": undefined,
      "nullableEnumValue": undefined,
      "nullableIntegerValue": undefined,
      "nullableNestedValue": undefined,
      "nullableNumberValue": undefined,
      "nullableObjectArrayValue": undefined,
      "nullablePrimitiveArrayValue": undefined,
      "nullableStringValue": undefined,
      "nullableUUIDValue": undefined,
      "optionalBooleanValue": undefined,
      "optionalDateStringValue": undefined,
      "optionalDateValue": undefined,
      "optionalEnumValue": undefined,
      "optionalIntegerValue": undefined,
      "optionalNestedValue": undefined,
      "optionalNumberValue": undefined,
      "optionalObjectArrayValue": undefined,
      "optionalPrimitiveArrayValue": undefined,
      "optionalStringValue": undefined,
      "optionalUUIDValue": undefined,
      "requiredBooleanValue": undefined,
      "requiredDateStringValue": undefined,
      "requiredDateValue": undefined,
      "requiredEnumValue": undefined,
      "requiredIntegerValue": undefined,
      "requiredNestedValue": undefined,
      "requiredNumberValue": undefined,
      "requiredObjectArrayValue": undefined,
      "requiredPrimitiveArrayValue": undefined,
      "requiredStringValue": undefined,
      "requiredUUIDValue": undefined,
      "stringValueWithDifferentWireName": undefined,
      "zeroBooleanValue": undefined,
    },
    "value": undefined,
  },
]
`;
